src/borg/archive.py

```diff
@@ -859,7 +859,17 @@ def process_file(self, path, st, cache, ignore_inode=False):
             with backup_io:
                 fh = Archive._open_rb(path)
             with os.fdopen(fh, 'rb') as fd:
-                self.chunk_file(item, cache, self.stats, backup_io_iter(self.chunker.chunkify(fd, fh)), compress=compress)
+                if is_special_file or st.st_size > self.chunker.min_size:
+                    self.chunk_file(item, cache, self.stats, backup_io_iter(self.chunker.chunkify(fd, fh)), compress=compress)
+                else:
+                    data = fd.read()
```

Comment:
the problem with avoiding the chunker is that you avoid the posix_fadvise also. so all these small files will blow up the cache.